### PR TITLE
Implement Building Videos From Image Sequences

### LINKED ISSUE #46: How to generate video from pictures?
Thank you for your Xabe.FFmpeg.
I not find generate video from pictures in document/Wiki .
Can you tell how to do it?

### PR DESCRIPTION
**Pull request by 06needhamt**
This PR Implements the ability to build videos from image sequences. This functionality is essentially the opposite of the `ExtractEveryNthFrame` function which extracts a sequence of images from a video.

Closes #46

### CODE REVIEW

File: Xabe.FFmpeg/Conversion/IConversion.cs [hunk 1]
```diff
@@ -54,11 +54,26 @@ public interface IConversion
...
         /// <returns>IConversion object</returns>
-        IConversion SetFrameRate(double frameRate);
+        IConversion BuildVideoFromImages(int startNumber, Func<string, string> buildInputFileName);
+
+        /// <summary>
+        /// Builds the -framerate option for the output of this conversion
+        /// </summary>
+        /// <param name="frameRate">The desired framerate of the output</param>
+        /// <returns>IConversion object</returns>
+        IConversion SetOutputFrameRate(double frameRate);
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg/Conversion/IConversion.cs]: We cannot rename methods in IConversion. It's already used and will break the contract. This require increate the major version of packages.

Comment (06needhamt) [reply Xabe.FFmpeg/Conversion/IConversion.cs]: I've renamed it back to SetFramerate

**Comment by tomaszzmuda** [comment]
This looks good but I'm worried about one think.
To the BuildVideoFromImages the user has to pass start number (it's fine) and inputBuilder.
What if user don't have correctly named images? If the number in file name is not 001 but 01. It will be hard to use. Everybody will have to prepare images to do this.
I'm thinking about passing there a list of image files. We will prepare them to conversion. It will be slower (probably require copying every image to Temp or something), but more user friendly.
What do you think about it?

**Comment by 06needhamt** [comment]
This sounds like a good idea, I will have a go at implementing it when I get chance hopefully sometime this weekend.

**Comment by TrueCP6** [comment]
If its possible could this feature also have the option to use the file selector in ffmpeg like `-i image%d.jpeg`?

**Comment by lkischler** [comment]
So maybe BuildVideoFromImages(IInputBuilder) and default implementation of IInputBuilder for file list? Then everyone can extends files selector.

**Comment by 06needhamt** [comment]
@lkischler Sounds like a good plan. One question though you write `IInputBuilder` as if you mean an entirely new interface, is this correct or are you referring to the `inputBuilder` delegate function I already added?

**Comment by 06needhamt** [comment]
I have created a `IInputBuilder` Interface with a default implementation that takes in a list of file paths, copies them to a Temp directory and renames them under the `img_xxx.xxx` scheme and returns a delegate function that can be passed in to `BuildVideoFromImages`.

**Comment by tomaszzmuda** [comment]
Noone will know that there is something like InputBuilder. The best options is to have 2 overloads of BuildVideoFromImages method. One like at is, and one with the IEnumerable<string> files
Conversion could use InputBuilder internally

**Comment by 06needhamt** [comment]
The startNumber Parameter has been removed

**Comment by tomaszzmuda** [comment]
Merged in #187 (I screw up with the merge) 
Nice work ;) I have to update documentation for this one

**Pull request closed without merging**